optimum/exporters/onnx/config.py

```diff
@@ -211,3 +212,88 @@ def generate_dummy_inputs_for_validation(self, reference_model_inputs: Mapping[s
+            behavior=behavior,
+        )
+
+        from ..tasks import TasksManager
+
+        if self._behavior is not ConfigBehavior.DECODER:
+            # retrieve the encoder config
+            encoder_onnx_config_constructor = TasksManager.get_exporter_config_constructor(
+                exporter="onnx", task="default", model_type=config.encoder.model_type
+            )
+            self._encoder_onnx_config = encoder_onnx_config_constructor(config.encoder)
+            self._normalized_config.ENCODER_NORMALIZED_CONFIG_CLASS = self._encoder_onnx_config._normalized_config
+
+        if self._behavior is not ConfigBehavior.ENCODER:
+            # retrieve the decoder config
+            decoder_onnx_config_constructor = TasksManager.get_exporter_config_constructor(
+                exporter="onnx", task="default", model_type=config.decoder.model_type
+            )
+            self._decoder_onnx_config = decoder_onnx_config_constructor(config.decoder, use_past=use_past)
+
+            self._normalized_config.DECODER_NORMALIZED_CONFIG_CLASS = self._decoder_onnx_config._normalized_config
+
+            if isinstance(self._decoder_onnx_config, OnnxSeq2SeqConfigWithPast):
+                self._past_key_values_generator = (
+                    DummySeq2SeqDecoderTextInputGenerator,
+                    DummySeq2SeqPastKeyValuesGenerator,
+                )
+            else:
+                self._past_key_values_generator = (
+                    DummySeq2SeqDecoderTextInputGenerator,
+                    DummyPastKeyValuesGenerator,
+                )
+
+            self.DUMMY_INPUT_GENERATOR_CLASSES += self._past_key_values_generator
+
+    @property
+    def torch_to_onnx_input_map(self) -> Mapping[str, str]:
+        if self._behavior is ConfigBehavior.DECODER:
+            return {
+                "decoder_input_ids": "input_ids",
+                "encoder_outputs": "encoder_hidden_states",
+                "attention_mask": "encoder_attention_mask",
+            }
+        return {}
+
+    def add_past_key_values(self, inputs_or_outputs: Mapping[str, Mapping[int, str]], direction: str):
+        return self._decoder_onnx_config.add_past_key_values(inputs_or_outputs, direction)
+
+    def flatten_past_key_values(self, flattened_output, name, idx, t):
+        return self._decoder_onnx_config.flatten_past_key_values(flattened_output, name, idx, t)
+
+    def flatten_output_collection_property(self, name: str, field: Iterable[Any]) -> Dict[str, Any]:
+        return self._decoder_onnx_config.flatten_output_collection_property(name, field)
+
+    def generate_dummy_inputs_for_validation(self, reference_model_inputs: Mapping[str, Any]) -> Mapping[str, Any]:
+        if self._behavior is ConfigBehavior.DECODER:
+            reference_model_inputs["input_ids"] = reference_model_inputs.pop("decoder_input_ids")
+            reference_model_inputs["encoder_hidden_states"] = reference_model_inputs.pop("encoder_outputs")[0]
```

Comment:
Why not use `self._decoder_onnx_config.generate_dummy_inputs_for_validation`?

Reply:
There can be decoders which inherited from `OnnxConfigWithPast` and would have the `dummy_inputs_for_validation` empty. We have to rewrite to handle those cases.